### ISSUE TITLE
davs2_git gives No working C compiler found error

### ISSUE THREAD

**Issue by kusaanko** [rdp/ffmpeg-windows-build-helpers#528]
The build commands are:

```bash
echo deb http://archive.ubuntu.com/ubuntu bionic main universe multiverse restricted | sudo tee /etc/apt/sources.list
echo deb http://security.ubuntu.com/ubuntu/ bionic-security main multiverse universe restricted | sudo tee -a /etc/apt/sources.list
echo deb http://archive.ubuntu.com/ubuntu bionic-updates main multiverse universe restricted | sudo tee -a /etc/apt/sources.list
git clone https://github.com/rdp/ffmpeg-windows-build-helpers.git
cd ffmpeg-windows-build-helpers
sudo apt-get update
sudo apt-get install subversion ragel curl texinfo g++ bison flex cvs yasm automake libtool autoconf gcc cmake git make pkg-config zlib1g-dev mercurial unzip pax nasm gperf autogen bzip2 autoconf-archive p7zip-full clang python3 python3-pip python3-setuptools python3-wheel ninja-build -y
if [[ "$(lsb_release -rs)" == *"18.04"* ]];then sudo apt install python3-distutils -y ;fi
if [[ "$(lsb_release -rs)" == *"20.04"* ]];then sudo apt install python-is-python3 -y ;fi
sudo pip3 install meson ninja
./cross_compile_ffmpeg.sh --build-ffmpeg-static=n --build-ffmpeg-shared=y --enable-gpl=y --compiler-flavors=win32 --build-intel-qsv=y --build-amd-amf=y
```

error

```
Cloning into 'davs2_git.tmp'...
remote: Enumerating objects: 1561, done.
remote: Total 1561 (delta 0), reused 0 (delta 0), pack-reused 1561
Receiving objects: 100% (1561/1561), 989.36 KiB | 1.09 MiB/s, done.
Resolving deltas: 100% (996/996), done.
done git cloning to davs2_git
doing git checkout origin/master
Note: checking out 'origin/master'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at b06d758 Fix decoder crash when decoder have decoding error with incorrect stream.
fetched no code changes, not forcing reconfigure for that...
configuring linux (/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/win32/davs2_git/build/linux) as $ PKG_CONFIG_PATH=/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32/lib/pkgconfig PATH=/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-i686/bin:$PATH ./configure --cross-prefix=/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32- --host=i686-w64-mingw32 --prefix=/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32 --enable-pic --disable-asm
all touch files already_configured* touchname= already_configured_76abd9638a22b7af98b80cbbfefc4ddb-
config options --cross-prefix=/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32- --host=i686-w64-mingw32 --prefix=/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32 --enable-pic --disable-asm ./configure
No working C compiler found.
```
Is there a way to skip this error?

**Comment by kusaanko** [comment]
And this is davs2's config.log

```
checking whether /home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-windres works... yes
checking whether !WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) is true... no
--------------------------------------------------
conftest.c:3:2: error: #error 
    3 | #error
      |  ^~~~~
--------------------------------------------------
Failed program was:
--------------------------------------------------
#include <winapifamily.h>
#if !(!WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)) 
#error  
#endif 
--------------------------------------------------
davs2 configure script
Command line options: "--cross-prefix=/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-" "--host=i686-w64-mingw32" "--prefix=/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32" "--enable-pic" "--disable-asm"

checking whether /home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-g++ works... no
Failed commandline was:
--------------------------------------------------
/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-g++ conftest.c -m32  -Wall -I. -I$(SRCPATH) -D_POSIX_C_SOURCE=200112L -DARCH_X86_64=0    -lshell32 -m32   -Wl,--large-address-aware -Wl,--dynamicbase,--nxcompat,--tsaware -o conftest
/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-i686/lib/gcc/i686-w64-mingw32/10.2.0/../../../../i686-w64-mingw32/bin/ld: cannot find -lstdc++
collect2: error: ld returned 1 exit status
--------------------------------------------------
Failed program was:
--------------------------------------------------
int main (void) {  return 0; }
--------------------------------------------------
DIED: No working C compiler found.
```

**Comment by rdp** [comment]
cannot find -lstdc++
Usually means the cross compiler didn't build all the way, nuke sandbox and
try again help?

On Thursday, December 31, 2020, Kusaanko <notifications@github.com> wrote:
> And this is davs2's config.log
>
> checking whether
/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-windres
works... yes
> checking whether !WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) is
true... no
> --------------------------------------------------
> conftest.c:3:2: error: #error
> 3 | #error
> | ^~~~~
> --------------------------------------------------
> Failed program was:
> --------------------------------------------------
> #include <winapifamily.h>
> #if !(!WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP))
> #error
> #endif
> --------------------------------------------------
> davs2 configure script
> Command line options:
"--cross-prefix=/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-"
"--host=i686-w64-mingw32"
"--prefix=/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32"
"--enable-pic" "--disable-asm"
>
> checking whether
/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-g++
works... no
> Failed commandline was:
> --------------------------------------------------
>
/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-g++
conftest.c -m32 -Wall -I. -I$(SRCPATH) -D_POSIX_C_SOURCE=200112L
-DARCH_X86_64=0 -lshell32 -m32 -Wl,--large-address-aware
-Wl,--dynamicbase,--nxcompat,--tsaware -o conftest
>
/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-i686/lib/gcc/i686-w64-mingw32/10.2.0/../../../../i686-w64-mingw32/bin/ld:
cannot find -lstdc++
> collect2: error: ld returned 1 exit status
> --------------------------------------------------
> Failed program was:
> --------------------------------------------------
> int main (void) { return 0; }
> --------------------------------------------------
> DIED: No working C compiler found.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.<
https://ci6.googleusercontent.com/proxy/SAS2s8KR5Hfm3LpUmhbfpzT_Uc4vOe89JlUZfB-qzL7O7O2q4hHCZY2MuXOVnRWHQ_LeGXCSQ5esoH0QhDovsFZxbhzf4Rl3R1wSiOIjfCZkQ2JdjuiKZd3C2IxQuOrtynI9F8EjbB5SOh8rfTXECebkKkBhh4X3--CMLK5Zr_TDQOsQyiKyUPaSIzj8QHtjM_l-HsrEHc-RTvhlppXEf71xW1kSnLx-uvlos6MKag=s0-d-e1-ft#https://github.com/notifications/beacon/AAADBUH5VWOTLRPFIW72RT3SXRBKZA5CNFSM4VONBSL2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOFTQGTHY.gif>

**Comment by kusaanko** [comment]
Removing the sandbox did not change the result.
However, reverting commit 9d26f71cb47d0b266439c99e42344baf0801e67d works fine.

**Comment by AsgeirSH** [comment]
I had this same issue, building on Ubuntu 18.04. Using master, but reverting commit 9d26f71 seems to have fixed this for me as well.

**Comment by kusaanko** [comment]
I ran it on Ubuntu 20.04, and the build was successful.
This problem seems to occur on Ubuntu 18.04.

**Comment by rdp** [comment]
Working OK here (20.04).  Do you have this file? `sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/lib/libstdc++.a`

**Comment by AsgeirSH** [comment]
I'll just swap to using Ubuntu 20.04 on my ffmpeg-build-VM, so this is not a problematic issue for me. I've cleaned all the build folders to build with the reverted commit, so I won't be able to check for that file in the failing configuration, sorry.

**Comment by kusaanko** [comment]
The libstdc++.a was not there when I ran it on Ubuntu 18.04.
I also found an error when I searched the build.log.
[build.log](https://github.com/rdp/ffmpeg-windows-build-helpers/files/5924103/build.log)

```
/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/pkgs/gcc/build/i686-w64-mingw32/libstdc++-v3/include/ext/pb_ds/hash_policy.hpp:52:10: fatal error: ext/pb_ds/detail/resize_policy/hash_load_check_resize_trigger_size_base.hpp: No such file or directory
   52 | #include <ext/pb_ds/detail/resize_policy/hash_load_check_resize_trigger_size_base.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
Makefile:1839: recipe for target 'i686-w64-mingw32/bits/extc++.h.gch/O2g.gch' failed
make[4]: *** [i686-w64-mingw32/bits/extc++.h.gch/O2g.gch] Error 1
make[4]: Leaving directory '/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/pkgs/gcc/build/i686-w64-mingw32/libstdc++-v3/include'
Makefile:563: recipe for target 'all-recursive' failed
make[3]: *** [all-recursive] Error 1
make[3]: Leaving directory '/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/pkgs/gcc/build/i686-w64-mingw32/libstdc++-v3'
Makefile:488: recipe for target 'all' failed
make[2]: *** [all] Error 2
make[2]: Leaving directory '/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/pkgs/gcc/build/i686-w64-mingw32/libstdc++-v3'
Makefile:11347: recipe for target 'all-target-libstdc++-v3' failed
make[1]: *** [all-target-libstdc++-v3] Error 2
make[1]: Leaving directory '/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/pkgs/gcc/build'
error

See build.log for more details.
Makefile:956: recipe for target 'all' failed
make: *** [all] Error 2
e/ext/pb_ds/hash_policy.hpp:52:10: fatal error: ext/pb_ds/detail/resize_policy/hash_load_check_resize_trigger_size_base.hpp: No such file or directory
   52 | #include <ext/pb_ds/detail/resize_policy/hash_load_check_resize_trigger_size_base.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
Makefile:1839: recipe for target 'i686-w64-mingw32/bits/extc++.h.gch/O2g.gch' failed
make[4]: *** [i686-w64-mingw32/bits/extc++.h.gch/O2g.gch] Error 1
make[4]: Leaving directory '/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/pkgs/gcc/build/i686-w64-mingw32/libstdc++-v3/include'
Makefile:563: recipe for target 'all-recursive' failed
make[3]: *** [all-recursive] Error 1
make[3]: Leaving directory '/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/pkgs/gcc/build/i686-w64-mingw32/libstdc++-v3'
Makefile:488: recipe for target 'all' failed
make[2]: *** [all] Error 2
make[2]: Leaving directory '/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/pkgs/gcc/build/i686-w64-mingw32/libstdc++-v3'
Makefile:11347: recipe for target 'all-target-libstdc++-v3' failed
make[1]: *** [all-target-libstdc++-v3] Error 2
make[1]: Leaving directory '/home/kusaanko/ffmpeg-windows-build-helpers/sandbox/cross_compilers/pkgs/gcc/build'
Makefile:956: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by AndCycle** [comment]
> 
> 
> Removing the sandbox did not change the result.
> However, reverting commit [9d26f71](https://github.com/rdp/ffmpeg-windows-build-helpers/commit/9d26f71cb47d0b266439c99e42344baf0801e67d) works fine.

I use gentoo,
got similar `No working C compiler found error` at ffmpeg compile, 
after playing around 9d26f71 gcc version `10.2.0` is the only reason why it failed in my environment.

**Comment by AndCycle** [comment]
ok, I found some clue around this missing `libstdc++.a` 
https://github.com/rdp/ffmpeg-windows-build-helpers/issues/259
I got similar error in log
```
pax: Unable to create gcc-10.2.0/libstdc++-v3/testsuite/tr1/7_regular_expressions/basic_regex/ctors/char/PaxHeaders.27018/: Is a directory
pax: Unable to create gcc-10.2.0/libstdc++-v3/testsuite/tr1/7_regular_expressions/basic_regex/ctors/char/PaxHeaders.27018/: Is a directory
pax: Unable to create gcc-10.2.0/libstdc++-v3/testsuite/tr1/7_regular_expressions/basic_regex/ctors/char/PaxHeaders.27018/: Is a directory
...
```

this issue closed in 1 day, the proper fix mentioned by https://github.com/rdp/ffmpeg-windows-build-helpers/issues/259#issuecomment-326807428 never get through.

**Comment by rdp** [comment]
Was that with 18.04?  Robert if there's a better version of gcc we could
use...

On Tuesday, February 9, 2021, AndCycle <notifications@github.com> wrote:
> ok, I found some clue around this missing libstdc++.a
> #259
> I got similar error in log
>
> pax: Unable to create
gcc-10.2.0/libstdc++-v3/testsuite/tr1/7_regular_expressions/basic_regex/ctors/char/PaxHeaders.27018/:
Is a directory
> pax: Unable to create
gcc-10.2.0/libstdc++-v3/testsuite/tr1/7_regular_expressions/basic_regex/ctors/char/PaxHeaders.27018/:
Is a directory
> pax: Unable to create
gcc-10.2.0/libstdc++-v3/testsuite/tr1/7_regular_expressions/basic_regex/ctors/char/PaxHeaders.27018/:
Is a directory
> ...
>
> this issue closed in 1 day, the proper fix mentioned by #259 (comment)
never get through.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.<
https://ci4.googleusercontent.com/proxy/DH4vt6qH_IbX-USuVE6085crvje_krQnoG_VzVDP-ZnRq9QE05Ruue3wEgiK3I8ZP1n6hUBC2VyAxYO3T_JWZ5UUSb_vxy26t4g2qx8qXxom8ajH86YH_Kg1HH-Lr32cd54roIFJK2biIDpmZV2Q_DQaIwDNNdQxH6Uca6F6DfHhT7zI4uisfV9h5x07R2e8mgh9Mr9Wz6M5EfLxZCYIt98zclHDfH65QQYKA1seJg=s0-d-e1-ft#https://github.com/notifications/beacon/AAADBUCZZZWGQ5PUVXL2G53S6IC3JA5CNFSM4VONBSL2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOFZDV7VI.gif>

**Comment by rdp** [comment]
patch to revert whatever the exact upgrade that causes this welcome...
